### PR TITLE
API phpunit 9 support

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10019

### CODE REVIEW

File: src/Search/Processors/SearchUpdateCommitJobProcessor.php [hunk 1]
```diff
@@ -232,7 +232,7 @@ protected function commitIndex($index)
         $this->completed[] = $name;
     }
 
-    public function setup()
+    public function setUp()
```
Comment (maxime-rainville) [inline src/Search/Processors/SearchUpdateCommitJobProcessor.php:235]: Revert this?

File: src/Search/Processors/SearchUpdateQueuedJobProcessor.php [hunk 2]
```diff
@@ -48,7 +48,7 @@ public function jobFinished()
         return $this->currentBatch >= count($this->batches);
     }
 
-    public function setup()
+    public function setUp()
```
Comment (maxime-rainville) [inline src/Search/Processors/SearchUpdateQueuedJobProcessor.php:51]: Revert this?

File: src/Solr/Reindex/Jobs/SolrReindexQueuedJobBase.php [hunk 3]
```diff
@@ -123,7 +123,7 @@ public function prepareForRestart()
         // NOOP
     }
 
-    public function setup()
+    public function setUp()
```
Comment (maxime-rainville) [inline src/Solr/Reindex/Jobs/SolrReindexQueuedJobBase.php:126]: Revert this?

File: tests/SolrIndexSubsitesTest.php [hunk 4]
```diff
@@ -221,6 +221,9 @@ public function testCorrectSubsiteIDOnPageWrite()
     public function testCorrectSubsiteIDOnFileWrite()
     {
         $subsiteIDs = ['0'] + $this->allFixtureIDs(Subsite::class);
+        $subsiteIDs = array_map(function ($v) {
```
Comment (maxime-rainville) [inline tests/SolrIndexSubsitesTest.php:224]: Can you clarify why this change is needed.

File: tests/SolrIndexSubsitesTest.php [hunk 5]
```diff
@@ -253,7 +256,7 @@ public function testCorrectSubsiteIDOnFileWrite()
             $this->assertCount(count($subsiteIDs), $mockWrite['statefulids']);
             foreach ($mockWrite['statefulids'] as $statefulIDs) {
                 $this->assertContains(
-                    $statefulIDs['state'][SearchVariantSubsites::class],
+                    (string) $statefulIDs['state'][SearchVariantSubsites::class],
```
Comment (maxime-rainville) [inline tests/SolrIndexSubsitesTest.php:259]: Please explain why this is needed

File: tests/SolrReindexTest.php [hunk 6]
```diff
@@ -259,7 +259,8 @@ public function testRunGroup()
 
         // Test that items in this variant / group are re-indexed
         // 120 divided into 6 groups should be 20 at least (max 21)
-        $this->assertEquals(21, count($ids), 'Group size is about 20', 1);
+        $c = count($ids);
+        $this->assertTrue($c === 20 || $c === 21, 'Group size is about 20', 1);
```
Comment (maxime-rainville) [inline tests/SolrReindexTest.php:263]: I think the `1` parameter was a `$delta` parameter, which doesn't exist anymore. You shouldn't need anymore since you converted the assertion to `assertTrue`.

```suggestion
        $this->assertTrue($c === 20 || $c === 21, 'Group size is about 20');
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
I think the renaming of the `setup` method to `setUp` was done my mistake. The [QueuedJob](https://github.com/symbiote/silverstripe-queuedjobs/blob/adb85fa17b7d9046e1cf31de13147946daea7d6f/src/Services/QueuedJob.php#L70) interface says it should be `setup`.

File: src/Search/Processors/SearchUpdateQueuedJobProcessor.php (hunk 2 above)
Comment (emteknetnz) [reply src/Search/Processors/SearchUpdateQueuedJobProcessor.php:51]: Updated

File: src/Solr/Reindex/Jobs/SolrReindexQueuedJobBase.php (hunk 3 above)
Comment (emteknetnz) [reply src/Solr/Reindex/Jobs/SolrReindexQueuedJobBase.php:126]: Updated

File: tests/SolrIndexSubsitesTest.php (hunk 4 above)
Comment (emteknetnz) [reply tests/SolrIndexSubsitesTest.php:224]: Ensure everything is the same type.  AssertContains function got stricter.

File: tests/SolrIndexSubsitesTest.php (hunk 5 above)
Comment (emteknetnz) [reply tests/SolrIndexSubsitesTest.php:259]: As above

File: tests/SolrReindexTest.php (hunk 6 above)
Comment (emteknetnz) [reply tests/SolrReindexTest.php:263]: Updated

File: src/Search/Processors/SearchUpdateQueuedJobProcessor.php (hunk 2 above)
Comment (maxime-rainville) [reply src/Search/Processors/SearchUpdateQueuedJobProcessor.php:51]: Not updated

File: tests/SearchUpdaterTest.php [hunk 7]
```diff
@@ -43,6 +43,7 @@ public function testBasic()
     {
         $item = new SearchUpdaterTest_Container();
         $item->write();
+        $this->assertTrue(true);
```
Comment (maxime-rainville) [inline tests/SearchUpdaterTest.php:46]: Damn these annoying tests that don't assert anything!

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
There's still one method that needs to be reverted to `setup`.

Everything else is fine.

**Comment by emteknetnz** [comment]
Have updated setup function

**Review by maxime-rainville — APPROVED** [review]
LGTM
